Blogpost: An update on the New Architecture Rollout (#3009)
facebook/react-native-website: website/blog/2022-03-14-an-update-on-the-new-architecture-rollout.md

---
title: An update on the New Architecture Rollout
authors: [cortinico]

> tags: [announcement]

Reviewer: ```suggestion
tags: [announcement]
date: 2022-03-15
```

Probably the date should be adjusted before publishing:
https://docusaurus.io/docs/next/api/plugins/@docusaurus/plugin-content-blog#date

To keep the discussion focused, we decided to have this working group **open to read** publicly and **restricted to write** only for approved users.

If you wish to join the conversation, you can [fill in this form](https://forms.gle/8emgdwFZXuzEpyyn9) to either **apply or nominate** someone that you think would be a valuable addition to the discussion.

**Everyone is welcome** to apply to join the conversation.

As every discussion forum, we would like to stress once more the importance of being **respectful** and welcoming towards others’ opinions. Please take the chance to read our [**code of conduct**](https://github.com/reactwg/react-native-new-architecture/blob/main/CODE_OF_CONDUCT.md) if you haven’t yet done it.

### The Migration Guide

After several rounds of review & feedback, we finally merged **the Migration Guide** (f.k.a. _the Playbook_). You can find it [on the website in the Guides section.](/docs/next/new-architecture-intro)

This Migration Guide will show you **how to create a custom Fabric component or a TurboModule** with a step-by-step approach. The guide will also show you how to **adapt your existing app or library** to use the New Architecture.

> Moreover, we would like to remind you the brand-new [Architecture section](/docs/next/architecture-overview) of our website. There you can find several in-depth articles and explanation of the React Native internals. Specifically, [the Fabric section](/docs/next/fabric-renderer) can help you understand the rendering pipeline in the New Architecture world.

Reviewer: ```suggestion
Moreover, we would like to remind you the brand-new [Architecture section](/architecture/overview) of our website. There you can find several in-depth articles and explanation of the React Native internals. Specifically, [the Fabric section](/architecture/fabric-renderer) can help you understand the rendering pipeline in the New Architecture world.
```

Links have changed after latest conversion of the architecture section page to unversioned docs.

<!-- alex ignore simple -->

In the next releases, we will keep on updating the template to make it even more streamlined and simple to use.

To enable the New Architecture on either platform, you can:

- On iOS, run `RCT_NEW_ARCH_ENABLED=1 pod install` inside the `ios` folder.
- On Android, set the `newArchEnabled` property to `true` by **either**:
  - Changing the corresponding line inside the `android/gradle.properties` file.
  - Set an environment variable `ORG_GRADLE_PROJECT_newArchEnabled=true`
  - Invoke Gradle with `-PnewArchEnabled=true`

Then you can **run your app** with `yarn react-native run-android` or `run-ios` and you’ll be running using Fabric and TurboModules enabled.

Please consider trying this new template, and [report any bug or unexpected behavior](https://github.com/reactwg/react-native-new-architecture/discussions/5) that you might face. Over the last months we worked hard to fix bugs and build failures that would have been **hard to catch** without the constant community feedback and testing.

### The 3rd-party Libraries Ecosystem

> We could not migrate to the New Architecture without the full support of **3rd-party libraries author & maintainers**.

Reviewer 2: ```suggestion
We can not migrate to the New Architecture without the full support of **3rd-party libraries author & maintainers**.
```
nit: this should not be using the past for "could", right? It's not like the new arch is out and everyone has migrated to it.

Or maybe it should be even more clear, along the lines of:
> The community won't be able to migrate to the New Architecture without the support of their fellow **libraries author & maintainers**.

Author: Fixed 👍

We understand how this can be a tedious process, and we understand the importance of supporting users on **both** old and New Architecture. Over the next months, we will focus on supporting our library developers to help them migrate over.

If you’re a **library developer**, [we invite you to post an update](https://github.com/reactwg/react-native-new-architecture/discussions/categories/libraries) in the New Architecture working group with the **status of your libraries**. This will help you attract early adopters and us to understand if any library is facing a blocker.

If instead you’re a **library user**, you can [post a message here](https://github.com/reactwg/react-native-new-architecture/discussions/6) to request a migration of a library. If we identify a library that becomes a blocker for a number of users, we will try to reach out to the maintainer and understand why they haven’t migrated yet.

> Finally, we would like to give a shout out to Software Mansion for releasing a new version of `[react-native-screens](https://github.com/software-mansion/react-native-screens)`, which has support for both architectures. Moreover, they published a blog-post ([Introducing Fabric to react-native-screens](https://blog.swmansion.com/introducing-fabric-to-react-native-screens-fd17bf18858e)) where they **tell their migration story**. We hope you will find this story inspiring and useful to tackle your migration.

Reviewer 2: ```suggestion
Finally, we would like to give a shout out to Software Mansion for releasing a new version of [`react-native-screens`](https://github.com/software-mansion/react-native-screens), which has support for both architectures. Moreover, they published a blog-post ([Introducing Fabric to react-native-screens](https://blog.swmansion.com/introducing-fabric-to-react-native-screens-fd17bf18858e)) where they **tell their migration story**. We hope you will find this story inspiring and useful to tackle your migration.
```
putting the `` the right place ;)

### Releases

Work on the 0.68 pre-release has realized much of [the improved release process we had defined last half](/blog/2022/01/19/version-067#improvements-to-release-process).

We’re happy to share that with 0.68 we were able to:

> - Successfully onboard release work to an internal rotation. Much of this is supported by [improved documentation](https://github.com/facebook/react-native/wiki/releases) on the release process which will reduce the bus factor of the release process.

Reviewer: ```suggestion
- Successfully onboard release work to an internal rotation. Much of this is supported by [improved documentation](/contributing/how-to-contribute) on the release process which will reduce the bus factor of the release process.
```

One more link updated, after wiki content move.

> - Initiated discussions with partners to support a [co-pilot rotation](https://github.com/facebook/react-native/wiki/Release-Roles-and-Responsibilities#release-role-2--release-copilot). We hope this effort will improve transparency of the process and inform our partners where to invest to support React Native releases and eco-system.

Reviewer: ```suggestion
- Initiated discussions with partners to support a [co-pilot rotation](/contributing/release-roles-responsibilities#release-role-2--release-copilot). We hope this effort will improve transparency of the process and inform our partners where to invest to support React Native releases and eco-system.
```

Wiki pages lately has been moved into website:
* https://reactnative.dev/contributing/release-roles-responsibilities#release-role-2--release-copilot

> - [Onboarded several Release Supporters and Testers from the community](https://github.com/reactwg/react-native-releases/discussions/11). We had put a call-out for help last half and so many folks stepped up! The feedback from our testers and supporters have **helped us fix crucial bugs** and regressions, especially around the new architecture, for the upcoming release. Thank you to everyone who signed up and tested out the release! If you’d like to join as a release tester or supporter, please sign up here!

Reviewer 2: ```suggestion
- [Onboarded several Release Supporters and Testers from the community](https://github.com/reactwg/react-native-releases/discussions/11). We had put a call-out for help last half and so many folks stepped up! The feedback from our testers and supporters have **helped us fix crucial bugs** and regressions, especially around the new architecture, for the upcoming release. Thank you to everyone who signed up and tested out the release! If you’d like to join as a release tester or supporter, [please sign up here](ADD ME)!
```
Link is missing

Author: Removing for now as the sign up is on hold for now (cc @lunaleaps).

Reviewer 3: https://forms.gle/fPuPE1MZRDGWNqpd6

We’re happy to share that with 0.68 we were able to:

- Successfully onboard release work to an internal rotation. Much of this is supported by [improved documentation](https://github.com/facebook/react-native/wiki/releases) on the release process which will reduce the bus factor of the release process.
- Initiated discussions with partners to support a [Copilot rotation](/contributing/release-roles-responsibilities#release-role-2--release-copilot). We hope this effort will improve transparency of the process and inform our partners where to invest to support React Native releases and eco-system.

> - [Onboarded several Release Supporters and Testers from the community](https://github.com/reactwg/react-native-releases/discussions/11). We had put a call-out for help last half and so many folks stepped up! The feedback from our testers and supporters have **helped us fix crucial bugs** and regressions, especially around the new architecture, for the upcoming release. Thank you to everyone who signed up and tested out the release! If you’d like to join as a release tester or supporter, please sign up here!

Reviewer 3: ```suggestion
- [Onboarded several Release Supporters and Testers from the community](https://github.com/reactwg/react-native-releases/discussions/11). We had put a call-out for help last half and so many folks stepped up! The feedback from our testers and supporters have **helped us fix crucial bugs** and regressions, especially around the new architecture, for the upcoming release. Thank you to everyone who signed up and tested out the release! If you’d like to join as a release tester or supporter, [please sign up here](https://forms.gle/fPuPE1MZRDGWNqpd6)!
```
